client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -282,14 +282,21 @@ public ClientStatistics clientStatistics() {
 
    @Override
    public ServerStatistics serverStatistics() {
+      return await(serverStatisticsAsync());
+   }
+
+   @Override
+   public CompletionStage<ServerStatistics> serverStatisticsAsync() {
       assertRemoteCacheManagerIsStarted();
       StatsOperation op = operationsFactory.newStatsOperation();
-      Map<String, String> statsMap = await(op.execute());
-      ServerStatisticsImpl stats = new ServerStatisticsImpl();
-      for (Map.Entry<String, String> entry : statsMap.entrySet()) {
-         stats.addStats(entry.getKey(), entry.getValue());
-      }
-      return stats;
+      return op.execute()
+            .thenApply(statsMap -> {
+               ServerStatisticsImpl stats = new ServerStatisticsImpl();
+               for (Map.Entry<String, String> entry : statsMap.entrySet()) {
```

Comment:
offtopic, why not change `StatsOperation` to return `ServerStatistics`? 
`ServerStatistics` creates a `HashMap` and here, we are creating another.

Reply:
It never occurred to me, I just moved the existing code to a lambda :)

Reply:
I changed StatsOperation